### ISSUE TITLE
[COMPILER REQUEST]: missing GCC 12.1 for some cross targets

### ISSUE THREAD

**Issue by dkm** [compiler-explorer/compiler-explorer#3729]
### Compiler name

gcc

### Compiler version

12.1.0

### Compiler language

c++,c,d,ada,go

### Compiler homepage

gcc.gnu.org

### Motivation

Missing for:
- avr (last 11.1.0)
- mips (last 11.2.0)
- mipsel (last 5.4.0)
- mips64 (last 11.2.0)
- mips64el (last 5.4.0)
- msp430 (last 4.5.3)
- kvx (last 9.3.0 -- but nothing more up to date is available)
- s390x (last 11.2.0)
- ppc (last 11.2.0)
- ppc64 (last 11.2.0)
- ppc64le (last 11.2.0)
- xtensa (last 8.4.0)
- esp32 (last 8.4.0)
- esp32s3 (last 8.4.0)

Not all target are possible, need to test one by one (same for the lang, most are c/c++ only)

**Comment by dkm** [comment]
- [ ] mips 12.1.0
  - [x] ct-ng build locally
  - [ ] build in docker
  - [ ] installed
  - [ ] infra
  - [ ] config change
- [ ] avr 12.1.0
  - [x] ct-ng build locally
  - [ ] build in docker
  - [ ] installed
  - [ ] infra
  - [ ] config change
- [ ] mipsel 12.1.0
  - [ ] ct-ng build locally
  - [ ] build in docker
  - [ ] installed
  - [ ] infra
  - [ ] config change
- [ ] mips64 12.1.0
  - [ ] ct-ng build locally
  - [ ] build in docker
  - [ ] installed
  - [ ] infra
  - [ ] config change
- [ ] mips64el 12.1.0
  - [ ] ct-ng build locally
  - [ ] build in docker
  - [ ] installed
  - [ ] infra
  - [ ] config change
- [ ] msp430 12.1.0
  - [ ] ct-ng build locally
  - [ ] build in docker
  - [ ] installed
  - [ ] infra
  - [ ] config change
- [ ] s390x 12.1.0
  - [ ] ct-ng build locally
  - [ ] build in docker
  - [ ] installed
  - [ ] infra
  - [ ] config change
- [ ] ppc 12.1.0
  - [x] ct-ng build locally
  - [ ] build in docker
  - [ ] installed
  - [ ] infra
  - [ ] config change
- [ ] ppc64 12.1.0
  - [ ] ct-ng build locally
  - [ ] build in docker
  - [ ] installed
  - [ ] infra
  - [ ] config change
- [ ] ppc64le 12.1.0
  - [ ] ct-ng build locally
  - [ ] build in docker
  - [ ] installed
  - [ ] infra
  - [ ] config change
- [ ] xtensa 12.1.0
  - [ ] ct-ng build locally
  - [ ] build in docker
  - [ ] installed
  - [ ] infra
  - [ ] config change
- [ ] esp32 12.1.0
  - [ ] ct-ng build locally
  - [ ] build in docker
  - [ ] installed
  - [ ] infra
  - [ ] config change
- [ ] esp32s3 12.1.0
  - [ ] ct-ng build locally
  - [ ] build in docker
  - [ ] installed
  - [ ] infra
  - [ ] config change

**Comment by dkm** [comment]
Needs hack around ct-ng to use gcc-12 (needed for cross-gnat). Will test it a bit more to be sure it's correct this time :)

**Comment by dkm** [comment]
TLDR; did some best effort. Not all lang/targets were easily built... Will do my best to add more

**Comment by mattgodbolt** [comment]
This is now live! Thanks @dkm
